packages/run-protocol/src/contractSupport.js

```diff
@@ -0,0 +1,55 @@
+// The proposal is not allowed to include any keys other than these,
+
+import { AmountMath } from '@agoric/ertp';
+
+const { details: X } = assert;
+
+/**
+ * Apply a delta to the `base` Amount, where the delta is represented as
+ * an amount to gain and an amount to lose. Typically one of those will
+ * be empty because gain/loss comes from the give/want for a specific asset
+ * on a proposal. We use two Amounts because an Amount cannot represent
+ * a negative number (so we use a "loss" that will be subtracted).
+ *
+ * @param {Amount} base
+ * @param {Amount} gain
+ * @param {Amount} loss
+ * @returns {Amount}
+ */
+export const applyDelta = (base, gain, loss) =>
+  AmountMath.subtract(AmountMath.add(base, gain), loss);
+
+// usually 'Collateral' and 'RUN'.
+export const assertOnlyKeys = (proposal, keys) => {
+  const onlyKeys = clause =>
+    Object.getOwnPropertyNames(clause).every(c => keys.includes(c));
+  assert(
+    onlyKeys(proposal.give),
+    X`extraneous terms in give: ${proposal.give}`,
+  );
+  assert(
+    onlyKeys(proposal.want),
+    X`extraneous terms in want: ${proposal.want}`,
+  );
+};
+
+/**
+ * Stage a transfer between `fromSeat` and `toSeat`, specified as the delta between
+ * the gain and a loss on the `fromSeat`. The gain/loss are typically from the
+ * give/want respectively of a proposal. The `key` is the allocation keyword.
+ *
+ * @param {ZCFSeat} fromSeat
+ * @param {ZCFSeat} toSeat
+ * @param {Amount} fromLoses
+ * @param {Amount} fromGains
+ * @param {Keyword} key
+ */
+export const transfer = (fromSeat, toSeat, fromLoses, fromGains, key) => {
```

Comment:
+1 to pulling this out in a shared module. Heads up there will be some merge conflicts after https://github.com/Agoric/agoric-sdk/pull/4877/ (e.g. `transfer` -> `stageDelta` atm)